app.rb

```diff
@@ -0,0 +1,79 @@
+require_relative 'person'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'nameable'
+require_relative 'trimmer_decorator'
+require_relative 'capitalize_decorator'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    puts 'List of all books:'
+    @books.each do |book|
+      puts "#{book.title} by #{book.author}"
+    end
+  end
+
+  def list_all_people
+    puts 'List of all people:'
+    @people.each do |person|
+      puts "#{person.id}, #{person.name}, #{person.age}"
+    end
+  end
+
+  def create_person(name, role, age, specialization)
+    if role == 'teacher'
+      person = Teacher.new(name, age, specialization)
+    elsif role == 'student'
+      person = Student.new(name, age)
+    else
+
+      return
+    end
+
+    @people << person
+    puts "#{person.name} (#{person.class}) has been created."
+  end
+
+  def create_book(title, author)
+    book = Book.new(title, author)
+    @books << book
+    puts "#{book.title} by #{book.author} has been added to the library."
+  end
+
+  def create_rental(person_id, book_title)
+    person = @people.find { |p| p.id == person_id }
+    book = @books.find { |b| b.title == book_title }
+
+    if person.nil?
+      puts "Person with ID #{person_id} not found."
+    elsif book.nil?
+      puts "Book with title '#{book_title}' not found."
+    else
+      rental = Rental.new(Time.now, book, person)
+      @rentals << rental
+      puts "#{book.title} has been rented by #{person.name} (#{person.class})."
+    end
```

Comment:
- [ ] While creating a rental you should provide `person` and `book` indexes not ID and you should provide  the `date` you are creating a rental

Reply:
resolved thanks